io/plugins/eu.esdihumboldt.hale.io.gml.test/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriterTest.java

```diff
@@ -809,7 +922,35 @@ private IOReport fillFeatureTest(String elementName, URI targetSchema,
 					Object value = geoms.iterator().next().getGeometry();
 
 					if (expected instanceof Geometry && value instanceof Geometry) {
-						matchGeometries((Geometry) expected, (Geometry) value);
+
+						// I have to comment below line to test Winding Order.
+						// Below method got assertion failed error.
+						if (windingOrderParam == null) {
+							matchGeometries((Geometry) expected, (Geometry) value);
+						}
+						// Winding Order Test.
+						if (windingOrderParam != null) {
+							if (windingOrderParam == EnumWindingOrderTypes.counterClockwise) {
+								assertFalse(((Geometry) expected).equalsExact((Geometry) value));
```

Comment:
This checks makes assumptions about the geometries passed in (that they are not counter-clockwise). I would remove it so there are no unnecessary (and undocumented) restrictions on this method. Just checking if the result complies with the rules is enough.